app/js/components/manageApps/ManageApps.jsx

```diff
@@ -131,6 +139,70 @@ export default class ManageApps extends React.Component {
     );
   }
 
+  fetchCurrentUser(){
+    const apiHelper = new ApiHelper(null);
+    const getUserData = new Promise(function(resolve, reject) {
+      apiHelper.get('/v1/session').then(response => {
+        resolve(response);
+      }).catch(error => {
+        toastr.error(error.message);
+      });
+    });
+    return getUserData;
+  }
+
+  fetchAdminProperties () {
+    const applicationDistribution = location.href.split('/')[2];
+    const urlPrefix = location.href.substr(0, location.href.indexOf('//'));
+    const url = location.href.split('/')[3];
+    const apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
+    const requestUrl = '/owa/allowModuleWebUpload';
+    axios.get(`${urlPrefix}/${apiBaseUrl}${requestUrl}`)
+      .then(response => {
+        this.setState((prevState, props) => {
+          return {
+            allowWebAdmin : response.data,
+          };
+        });
+      })
+      .catch(error => {
+        toastr.error(error.message);
+      });
+  }
+
+  checkIsAdmin() {
+    this.fetchCurrentUser().then((response) => {
+      const user = response.user;
+      if (user) {
+        let userPrivileges = [];
+        let userCanManageModules;
+        userPrivileges = userPrivileges.concat(user.privileges);
+        if (userPrivileges.length > 0) {
+          const managePrivilege = userPrivileges.find((managemodules) => {
+            if (managemodules.display === "Manage Modules") {
+              return userCanManageModules = true;
+            }
+            return userCanManageModules = false;
+          });
+        }
+        if (user.systemId === "admin" || user.display === "admin" || userCanManageModules){
```

Comment:
"admin" is just a value, there is no requirement for a system admin to have a system id of any value. It all depends on the System Admin role, irrespective of the user name or system id.

Reply:
so I should change using the system id ???

Reply:
@dkayiwa

Reply:
Am not talking about using system id. Am talking about is that can't an admin be samuel?

Reply:
so if one is an admin and the name is different it will check if they have the priviledges.

Reply:
Why do you even have to assume any name called admin?

Reply:
Or what happens if my name or system id is admin but am not an administrator?

Reply:
I was checking on the response of the rest and if the user is an admin the system id is listed as admin.

Reply:
I have added an image of the response.